tests/models/bark/test_modeling_bark.py

```diff
@@ -1022,26 +1052,30 @@ def test_generate_end_to_end_with_sub_models_args(self):
         input_ids = self.inputs
 
         with torch.no_grad():
+            torch.manual_seed(0)
             self.model.generate(
                 **input_ids, do_sample=False, temperature=1.0, coarse_do_sample=True, coarse_temperature=0.7
             )
-            self.model.generate(
+            output_ids_without_min_eos_p = self.model.generate(
                 **input_ids,
-                do_sample=False,
-                temperature=1.0,
+                do_sample=True,
+                temperature=0.9,
                 coarse_do_sample=True,
                 coarse_temperature=0.7,
                 fine_temperature=0.3,
             )
-            self.model.generate(
+
+            output_ids_with_min_eos_p = self.model.generate(
                 **input_ids,
                 do_sample=True,
-                temperature=0.6,
-                penalty_alpha=0.6,
-                semantic_temperature=0.9,
-                coarse_temperature=0.2,
-                fine_temperature=0.1,
+                temperature=0.9,
+                coarse_temperature=0.7,
+                fine_temperature=0.3,
+                min_eos_p=0.1,
             )
+        self.assertLess(
+            len(output_ids_with_min_eos_p[0, :].tolist()), len(output_ids_without_min_eos_p[0, :].tolist())
+        )
```

Comment:
Nice ! Thanks for iterating here! 

let's also test equality between `output_ids_with_min_eos_p[:,len_without_min_eos]` and `output_ids_without_min_eos_p`! then we are good to merge!

Reply:
Doesn't `model.generate()` give an audio array and not token IDs? I'm quite sure how to explain the discrepancies I'm seeing: The generated audios with and without the parameter up till the end of the one with do not match. Even the first elements do not match:

```
======================================================================
FAIL: test_generate_end_to_end_with_sub_models_args (tests.models.bark.test_modeling_bark.BarkModelIntegrationTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/nfs/isaacchung/work/transformers/tests/models/bark/test_modeling_bark.py", line 1080, in test_generate_end_to_end_with_sub_models_args
    self.assertEqual(output_ids_with_min_eos_p[0, :].tolist()[0], output_ids_without_min_eos_p[0, :].tolist()[0])
AssertionError: -0.014097243547439575 != 0.03543265536427498
```


Details
----
Asserting equality for the list took a long time (1hr+). In comparison, calling `model.generate()` twice takes around 1min. With these asserts and a torch.manual_seed set:
```
len_output_ids_with_min_eos_p = len(output_ids_with_min_eos_p[0, :].tolist())
self.assertLess(
    len_output_ids_with_min_eos_p, len(output_ids_without_min_eos_p[0, :].tolist())
)
self.assertListEqual(
    output_ids_with_min_eos_p[0, :].tolist(),
    output_ids_without_min_eos_p[0, :len_output_ids_with_min_eos_p].tolist(),
)
```
we get this failure consistently (ran a few times, thus the delay):
```
======================================================================
FAIL: test_generate_end_to_end_with_sub_models_args (tests.models.bark.test_modeling_bark.BarkModelIntegrationTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/nfs/isaacchung/work/transformers/tests/models/bark/test_modeling_bark.py", line 1080, in test_generate_end_to_end_with_sub_models_args
    self.assertListEqual(
AssertionError: Lists differ: [-0.014097243547439575, -0.015964325517416[1268758 chars]6505] != [0.03543265536427498, 0.03593989461660385,[1260898 chars]7222]

First differing element 0:
-0.014097243547439575
0.03543265536427498

Diff is 2883903 characters long. Set self.maxDiff to None to see it.

----------------------------------------------------------------------
Ran 1 test in 3876.261s

FAILED (failures=1)
```

We know that the `with` size is indeed less than the `without` size:
```
torch.Size([1, 58560]) != torch.Size([1, 62720])
```
and the seed works as this size check gives the same outputs over 3 runs.

Here are the generated files within the same test run with the same seed:
[Archive.zip](https://github.com/huggingface/transformers/files/13179495/Archive.zip)

Reply:
Oh you're right about the waveform, difficult to test this. Better do what I recommended for the `test_generate_semantic_early_stop` test then ! Should be much faster! And for this test, your verification are more than good!

Reply:
Good call! I've also followed how `test_generate_semantic` asserts the first 10 tokens. Ran the integration tests again 👍 ✅